File: update_tailwind.js
const fs = require('fs');
const path = require('path');

const adminFile = path.join(__dirname, 'admin_dashboard', 'code.html');
const memberDir = path.join(__dirname, '..', 'Member');

const adminHtml = fs.readFileSync(adminFile, 'utf8');

// Extract the tailwind config block
const match = adminHtml.match(/(<script id="tailwind-config">[\s\S]*?<\/script>)/);
if (!match) {
    console.error("Could not find tailwind config in admin file.");
    process.exit(1);
}

const adminConfig = match[1];

let count = 0;
const files = fs.readdirSync(memberDir);
for (const filename of files) {
    if (filename.endsWith('.html')) {
        const filepath = path.join(memberDir, filename);
        const content = fs.readFileSync(filepath, 'utf8');
        
        // Replace the existing tailwind config
        const newContent = content.replace(/<script id="tailwind-config">[\s\S]*?<\/script>/g, adminConfig);
        
        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf8');
            console.log(`Updated ${filename}`);
            count++;
        } else {
            console.log(`No config block found in ${filename}`);
        }
    }
}

console.log(`Updated ${count} files.`);

File: update_tailwind.py
import os
import re

admin_file = r"admin_dashboard\code.html"
member_dir = r"..\Member"

with open(admin_file, 'r', encoding='utf-8') as f:
    admin_html = f.read()

# Extract the tailwind config block
match = re.search(r'(<script id="tailwind-config">.*?</script>)', admin_html, re.DOTALL)
if not match:
    print("Could not find tailwind config in admin file.")
    exit(1)

admin_config = match.group(1)

count = 0
for filename in os.listdir(member_dir):
    if filename.endswith(".html"):
        filepath = os.path.join(member_dir, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Replace the existing tailwind config
        new_content, num_subs = re.subn(r'<script id="tailwind-config">.*?</script>', admin_config, content, flags=re.DOTALL)
        
        if num_subs > 0:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated {filename}")
            count += 1
        else:
            print(f"No config block found in {filename}")

print(f"Updated {count} files.")
